### PR TITLE
No password requirement anymore on backend, new checklist format.

### PR DESCRIPTION
**Pull request by andrewconner**
@leogrim

### CODE REVIEW

**Comment by andrewconner** [comment]
@eishay

**Comment by andrewconner** [comment]
Not sure who should review this. Manually tested in 4 cases:
1. Email signup with password works as expected
2. Email signup sending no password auto-assigns password
3. Facebook signup auto-assigns password
4. Twitter signup auto-assigns password

File: kifi-backend/modules/shoebox/app/com/keepit/controllers/core/AuthController.scala [hunk 1]
```diff
@@ -270,7 +270,7 @@ class AuthController @Inject() (
         authHelper.checkForExistingUser(info.email) collect {
           case (emailIsVerifiedOrPrimary, sui) if sui.credentials.isDefined && sui.userId.isDefined =>
             val identity = sui.credentials.get
-            val matches = hasher.matches(identity.passwordInfo.get, new String(info.password))
+            val matches = hasher.matches(identity.passwordInfo.get, info.password)
```
Comment (leogrim) [inline kifi-backend/modules/shoebox/app/com/keepit/controllers/core/AuthController.scala:273]: woohoo!

**Comment by leogrim** [comment]
:+1:

**Pull request merged**